baseplate/__init__.py

```diff
@@ -539,21 +539,27 @@ def configure_context(self, *args: Any, **kwargs: Any) -> None:  # noqa: F811
         cfg = config.parse_config(app_config, context_spec)
         self._context_config.update(cfg)
 
-    def add_to_context(
-        self, name: str, context_factory: "baseplate.clients.ContextFactory"
-    ) -> None:
-        """Add an attribute to each request's context object.
+    def add_to_context(self, name: str, attribute_config: Any) -> None:
+        """Add an attribute or a structure of attributes to each request's context object.
+
+        The given attribute config object can be one of the following::
+            - An arbitrary object to be added to the :py:class:`~baseplate.RequestContext`.
+
+            - A factory with a method named ``make_object_for_context``.
+            On each request, the factory will be asked to create an appropriate
+            object to attach to the :py:class:`~baseplate.RequestContext`.
 
```

Comment:
This reST syntax is leading to some kinda funky output when rendered:

![Screenshot from 2020-10-20 11-10-33](https://user-images.githubusercontent.com/338853/96626894-ee8c6380-12c4-11eb-916d-9ca1986218d3.png)

How about something like:

```suggestion
        The given attribute config object can be one of the following:        
                                                                              
        * An arbitrary object to be added to the                              
        :py:class:`~baseplate.RequestContext`.                                
                                                                              
        * A factory with a method named ``make_object_for_context``.  On each 
        request, the factory will be asked to create an appropriate object to 
        attach to the :py:class:`~baseplate.RequestContext`.                  
                                                                              
        * A dict containing arbitrary objects, factories, or other dicts.  In 
        this case, a nested object will be added to the context. Each item of 
        the dict will be processed using the same rules to become an attribute
        of the nested object.                                                 

```

(you can test this locally in OneVM with `make docs` and visiting https://baseplate-py.reddit.local)

Reply:
Yes, this looks much better. Thanks for the suggestion!